src/plugins/data/public/query/saved_query/saved_query_service.ts

```diff
@@ -112,8 +112,12 @@ export const createSavedQueryService = (
   };
 
   const getSavedQuery = async (id: string): Promise<SavedQuery> => {
-    const response = await savedObjectsClient.get<SerializedSavedQueryAttributes>('query', id);
-    return parseSavedQueryObject(response);
+    const savedObject = await savedObjectsClient.get<SerializedSavedQueryAttributes>('query', id);
```

Comment:
non blocker for this PR, but: should the services be showing toasts or should we rather return an error and the application using it can decide whether it wants to show a toast or handle the error in some other way.

Reply:
@ppisljar FYI I'm migrating this part into https://github.com/elastic/kibana/pull/56643